### PR TITLE
add instrumentation for delayed job

### PR DESCRIPTION
**Pull request by mbobin**
Hello!

This is my attempt for the delayed_job metrics #2 issue. I'm not really convinced that `instrumentation/delayed_job` is the right place for the plugin definition, but it enables this kind of usage:

```ruby
require 'prometheus_exporter/instrumentation'
PrometheusExporter::Instrumentation::DelayedJob.register_plugin
```

Let me know what you think.

### CODE REVIEW

File: lib/prometheus_exporter/instrumentation/delayed_job.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      start = ::Process.clock_gettime(::Process::CLOCK_MONOTONIC)
+      result = block.call(job, *args)
+      success = true
+      result
+    ensure
+      duration = ::Process.clock_gettime(::Process::CLOCK_MONOTONIC) - start
+
+      @client.send_json(
+        type: "delayed_job",
+        name: YAML.load(job.handler).job_data["job_class"].to_s,
```
Comment (SamSaffron) [inline lib/prometheus_exporter/instrumentation/delayed_job.rb]: oh do we need this whole yaml load thing just to get the job class?

**Comment by SamSaffron** [comment]
This looks fine to me. 

Only a few minor changes

- Can you also update the README with instructions? 
- Can you include a test or 2? (I am not sure if I set the example properly with sidekiq, but we should have tests here) 

I think `PrometheusExporter::Instrumentation::DelayedJob.register_plugin` is fine, not sure where else we can put it

File: lib/prometheus_exporter/instrumentation/delayed_job.rb (hunk 1 above)
Comment (mbobin) [reply lib/prometheus_exporter/instrumentation/delayed_job.rb]: You're right. A regexp would do a better job.

**Comment by SamSaffron** [comment]
looking good, we just need the readme updated!

**Comment by SamSaffron** [comment]
Thanks heaps, failure is travis being super annoying and refusing to work with ipv6

**Pull request merged**